File: 2614. Prime In Diagonal/solution.py
class Solution(object):
  def diagonalPrime(self, nums):
    """
    :type nums: List[List[int]]
    :rtype: int
    """
    result = 0
    for index in range(len(nums)):
      diagonal = nums[index][index]
      diagonalPrime = True
      if diagonal < 2:
        diagonalPrime = False
      else:
        squaredDiagonal = int(sqrt(diagonal))
        for index2 in range(2, squaredDiagonal + 1):
          if diagonal % index2 == 0:
            diagonalPrime = False
            break
      if diagonalPrime and diagonal > result:
          result = diagonal
      
      secondDiagonal = nums[len(nums) - index - 1][index]
      secondDiagonalPrime = True
      if secondDiagonal < 2:
        secondDiagonalPrime = False
      else:
        squaredSecondDiagonal = int(sqrt(secondDiagonal))
        for index2 in range(2, squaredSecondDiagonal + 1):
          if secondDiagonal % index2 == 0:
            secondDiagonalPrime = False
            break
      if secondDiagonalPrime and secondDiagonal > result:
          result = secondDiagonal
    return result

File: 2614. Prime In Diagonal/solution.cpp
class Solution {
public:
  int diagonalPrime(vector<vector<int>>& nums) {
    int result = 0;
    for (int index = 0; index < nums.size(); index++){
      int diagonal = nums[index][index];
      bool diagonalPrime = true;
      if (diagonal < 2){
        diagonalPrime = false;
      }
      else{
        int squaredDiagonal = sqrt(diagonal);
        for (int index2 = 2; index2 < squaredDiagonal + 1; index2++){
          if (diagonal % index2 == 0){
            diagonalPrime = false;
            break;
          }
        }
      }
      if (diagonalPrime && diagonal > result){
          result = diagonal;
      }
      
      int secondDiagonal = nums[nums.size() - index - 1][index];
      bool secondDiagonalPrime = true;
      if (secondDiagonal < 2){
        secondDiagonalPrime = false;
      }
      else{
        int squaredSecondDiagonal = sqrt(secondDiagonal);
        for (int index2 = 2; index2 < squaredSecondDiagonal + 1; index2++){
          if (secondDiagonal % index2 == 0){
            secondDiagonalPrime = false;
            break;
          }
        }
      }
      if (secondDiagonalPrime && secondDiagonal > result){
        result = secondDiagonal;
      }
    }
    return result;
  }
};